framework/update/helper/updateui_win.cpp

```
@@ -0,0 +1,642 @@
+    }
+
+    MSG message = { };
+    while (::GetMessageW(&message, nullptr, 0, 0) > 0) {
+        ::TranslateMessage(&message);
+        ::DispatchMessageW(&message);
+    }
+
+    //! NOTE: Normally the reader has already finished - it is what asked the
+    //! window to close. It is still sitting in ReadFile if the window was
+    //! closed by hand, and cancelling that is what lets it end.
+    ::CancelSynchronousIo(thread);
```

Clang (14.0.6)

[warning] variable 'thread' is not initialized

(cppcoreguidelines-init-variables)